### PR TITLE
Make unrealized `SingleQubitQPDGate` have definition of `None` rather than raise error

### LINKED ISSUE #417: Serializing/Deserializing a circuit with QPD gates causes a ValueError
`ValueError: Missing \'basis_id\': unable to realize SingleQubitQPDGate` is raised when serializing/deserializing a circuit with `QPDGate`s.

This should not raise an error. The gate should be able to be serialized and deserialized without being decomposed into single-qubit Qiskit gates.

### PR DESCRIPTION
**Pull request by garrison**
Fixes #417 according to https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/issues/417#issuecomment-1783385101

Needs

- [x] release note
- [x] test based on https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/issues/417#issuecomment-1783096682

### CODE REVIEW

File: circuit_knitting/cutting/qpd/instructions/qpd_gate.py [hunk 1]
```diff
@@ -101,11 +99,6 @@ def __eq__(self, other):
             and self.label == other.label
         )
 
-    @abstractmethod
-    def _define(self) -> None:
-        """Generate a decomposed gate."""
-        raise NotImplementedError  # pragma: no cover
-
```
Comment (garrison) [inline circuit_knitting/cutting/qpd/instructions/qpd_gate.py:108]: Removing these lines wasn't essential, but seems more in line with the idea of relying on the base class's `Instruction._define` [to do nothing rather](https://github.com/Qiskit/qiskit/blob/e181eabc04252d332448f49399de9b72ab86b375/qiskit/circuit/instruction.py#L269-L271) than raise an error if a definition isn't explicitly provided.

**Comment by garrison** [comment]
Come to think of it, making this change _might_ allow us to avoid having to override the `_directive` property, as well.

https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/2df09335477469c949cf714f7cf6210cc40ebdf8/circuit_knitting/cutting/qpd/instructions/qpd_gate.py#L210-L213

File: circuit_knitting/cutting/qpd/instructions/qpd_gate.py (hunk 1 above)
Comment (Ibrahim-Shehzad) [reply circuit_knitting/cutting/qpd/instructions/qpd_gate.py:108]: This makes sense to me, thanks.

**Comment by Ibrahim-Shehzad** [comment]
This resolution looks good to me, thank you.

**Review by Ibrahim-Shehzad — APPROVED** [review]
This resolution makes sense, thank you.

**Review by caleb-johnson — DISMISSED** [review]
LGTM thanks 👍

**Pull request merged**